Scripts/main.py

```diff
@@ -0,0 +1,235 @@
+    """ The issameorientation function returns a bool returning true if a pair of reads have the same orientation and
+    false if they have an oposite orientation.
+
+    :param read: Pysam object containing data of a read.
+    :return bool: A boolean returning True if the reads of a pair have the same orientation.
+    """
+    if read.is_paired and not read.mate_is_unmapped:
+        if read.is_reverse and read.mate_is_reverse:
+            return True
+
+        elif not read.is_reverse and not read.mate_is_reverse:
+            return True
+
+        else:
+            return False
+
+
+def write_bedfile(regions, read_data):
+    """ The write_bedfile function writes a file in BED format that can be loaded in igv and visualises the read data.
+
+    :param regions: a list of coordinates specified in the vcf file.
+    :param read_data: a 2d list containing the read data of every region.
+    """
+    text = 'track name=CNV_information description="Region_Summary." db=hg19 gffTags=on\n'
+
+    for index in range(0, len(regions)):
+        region = f"{regions[index][0]}\t{regions[index][1]}\t{regions[index][2]}"
+        paired_reads = read_data[index][0]
+        unmapped_mate = read_data[index][1]
+        duplicate_pairs = read_data[index][2]
+        high_insize = read_data[index][3]
+        facaway = read_data[index][4]
+        same_orientation = read_data[index][5]
+        proper_pair = read_data[index][6]
+
+        text += f"{region}\tName=Read_information;Paired_reads={paired_reads};Proper_pairs={proper_pair};Unmapped_mate=" \
+                f"{unmapped_mate};Duplicate_pair={duplicate_pairs};High_insert_size={high_insize};Facaway={facaway};" \
+                f"Same_orientation={same_orientation}\n"
+
+    with open(args.output + '/output.bed', 'w') as igvfile:
+        igvfile.write(text)
+
+
+def write_logfile():
+    """ The write logfile function writes a log.txt file in the output folder and writes all the parameters down."""
+    current_path = os.getcwd()
+
+    current_time = datetime.now().strftime("%H:%M:%S")
+    current_day = date.today().strftime("%d/%m/%Y")
+
+    text = f'Logfile created by: {current_path}\nScript finished at: {current_time} {current_day}\n{"-"*80}\n' \
+           f'Parameters:\nBamfile: {args.bamfile}\nVCF file: {args.vcf_file}\nOutput_folder: {args.output}\n' \
+           f'Capture region: {args.capture_region}\nHigh insert size threshold: {args.high_insert_size}'
+
+    with open(args.output + '/log.txt', 'w') as logfile:
+        logfile.write(text)
+
+
+if __name__ == '__main__':
+    """ The main calls the other functions in the right order."""
+
+    regions = vcf_calls()  # get coordinates of the regions specified in vcf file.
+
+    all_reads = fetch_reads(regions)  # get all reads at the regions from the vcf call.
+
+    read_data = get_read_data(all_reads)  # retrieve data from reads.
+
+    write_bedfile(regions, read_data)  # write the result in a BED file.
```

Comment:
Is het qua geheugen gebruik niet slimmer om per regio de functies all_reads, read_data en write_bedfile uit te voeren? Dat zou je dan zelfs kunnen parallelliseren mocht dat nodig zijn.